nncf/data/generators.py

```diff
@@ -13,13 +13,22 @@
 
 import nncf
 from nncf.common.logging.track_progress import track
+from nncf.common.telemetry_extractors import DatasetGeneratedFromApi
+from nncf.telemetry import tracked_function
+from nncf.telemetry.events import NNCF_COMMON_CATEGORY
 
 BASE_VOCAB_SIZE = 12000
 
 TModel = TypeVar("TModel")
 TTokenizer = TypeVar("TTokenizer")
 
 
+@tracked_function(
+    category=NNCF_COMMON_CATEGORY,
+    extractors=[
+        DatasetGeneratedFromApi(),
+    ],
+)
 def generate_text_data(
```

Comment:
Please mark this function with the api decorator.

Reply:
Updated.